### PR TITLE
Add live testing of Postera uploading

### LINKED ISSUE #633: Make ManifoldArtifacts class use a schema.
sorry if you've been over this already but why aren't we using a pydantic schema here? instead of need an init method and saving everything to the object here

_Originally posted by @apayne97 in https://github.com/choderalab/asapdiscovery/pull/595#discussion_r1396223136_

### LINKED ISSUE #686: Live push/pull to postera in sandboxed test account.
Postera folks issued us a new API key just for testing. We can do live push pull tests with this API key without clogging the workspace of the medchemists and should use this.

### LINKED ISSUE #700: Possible postera edge case where molecule with undefined stereo is docked.
I need to double check this, but running a docking ticket from postera with some undefined stereo has been giving me grief, will confirm here.

### PR DESCRIPTION
**Pull request by hmacdope**
## Description
Fixes #700 
Fixes #686
Fixes #633

## Developers certificate of origin
- [X] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by hmacdope** [comment]
@JenkeScheen note that this PR changes the asapdiscovery postera API slightly, let me know if anything you are using stops working once this is merged and I can help you update it.

**Comment by hmacdope** [comment]
@dotsdl @jthorton @mikemhenry would I be able to snag a review here? Need to get this one in soonish to close of 0.2.2 and finalise #832

File: asapdiscovery-data/asapdiscovery/data/postera/manifold_artifacts.py [hunk 1]
```diff
@@ -29,90 +32,81 @@ class ArtifactType(Enum):
...
+    molecule_set_id: Optional[str] = Field(
+        None, description="The UUID of the molecule set to upload to"
+    )
+    molecule_set_name: Optional[str] = Field(
+        None, description="The name of the molecule set to upload to"
+    )
+
+    bucket_name: str = Field(..., description="The name of the bucket to upload to")
+
+    artifact_columns: list[str] = Field(
```
Comment (jthorton) [inline asapdiscovery-data/asapdiscovery/data/postera/manifold_artifacts.py:55]: Should the type include optional if the default is None?

Comment (jthorton) [reply asapdiscovery-data/asapdiscovery/data/postera/manifold_artifacts.py:55]: A couple of the other fields also seem to have None as the default value but its not a expected type.

Comment (hmacdope) [reply asapdiscovery-data/asapdiscovery/data/postera/manifold_artifacts.py:55]: Those are designed to throw if nothing is passed, ie there is no sensible default value it must be provided. Is there a better way to acheive that?

Comment (jthorton) [reply asapdiscovery-data/asapdiscovery/data/postera/manifold_artifacts.py:55]: Ah right so would `...` work?

Comment (hmacdope) [reply asapdiscovery-data/asapdiscovery/data/postera/manifold_artifacts.py:55]: No, it should throw I think as will fail the validation.

**Review by jthorton — APPROVED** [review]
Few small cleanups feel free to merge after! Great job with the testing!
